src/utils/messages.js

```diff
@@ -1,20 +1,28 @@
-const generateMessage = (username, text) => {
-    return {
-        username,
-        text,
-        createdAt: new Date().getTime(),
-    }
-}
+const User = require('../entities/user')
 
-const generateLocationMessage = (username, url) => {
-    return {
-        username,
-        url,
-        createdAt: new Date().getTime(),
-    }
-}
+const admin = new User({ username: 'admin' })
+
+const makeMessageGenerator =
+    (author, process = (p) => p, payloadKey = 'text') =>
```

Comment:
Could you please clarify the `process = (p) => p` parameter of this function?

Reply:
it is default message payload processor, it does nothing returning a passed value